### PR TITLE
spaghetty plot

### PR DESCRIPTION
**Pull request by abuts**
This PR fixes issues with spaghetti_plot identified by user.

Need unit tests for `spaghetti_plot`

### CODE REVIEW

File: horace_core/spaghetti_plot.m [hunk 1]
```diff
@@ -186,17 +186,17 @@
...
+elseif szq == nseg  % "Square" for each segment
     qwidth = repmat(opt.qwidth(:), 2, 1);
-  case [2, nseg]              % Rectangular for each segments
+elseif szq == 2*nseg % Rectangular for each segments
     qwidth = reshape(opt.qwidth, 2, nseg);
-  otherwise
+else
     error('HORACE:spaghetti_plot:invalid_argument', ...
           ['qwidth size must be one of: [1, 1], [2, 1], [1, nseg], or [2, nseg].\n' ...
            'Received: %s'], disp2str(opt.qwidth))
```
Comment (oerc0122) [inline horace_core/spaghetti_plot.m:202]: This is fundamentally wrong and will cause further errors
```suggestion
siz = size(opt.qwidth);

if isequal(siz, [1, 1])                % "Square" for all segments
    qwidth = repmat(opt.qwidth, 2, nseg);
elseif isequal(siz,  [1, 2]) || ...    % Rectangular for all segments
        isequal(siz, [2, 1])
    qwidth = repmat(opt.qwidth(:), 1, nseg);
elseif isequal(siz,  [nseg, 1]) || ... % "Square" for each segment
        isequal(siz, [1, nseg])
    qwidth = repmat(opt.qwidth(:)', 2, 1);
elseif isequal(siz,  [2, nseg])        % Rectangular for each segments
    qwidth = reshape(opt.qwidth, 2, nseg);
else
    error('HORACE:spaghetti_plot:invalid_argument', ...
          ['qwidth size must be one of: [1, 1], [2, 1], [1, nseg], or [2, nseg].\n' ...
           'Received: %s'], disp2str(opt.qwidth))
```

Comment (abuts) [reply horace_core/spaghetti_plot.m:202]: Have not found any errors, so do not see what is wrong with this. The suggested change fails with error
```SWITCH expression must be a scalar or a character vector.

Error in spaghetti_plot (line 190)
```
on the version of the Matlab installed in IDAAAS

Comment (oerc0122) [reply horace_core/spaghetti_plot.m:202]: There is a significant difference between an array with 12 elements and a 4x3 array. Particularly where repmat is concerned.

Comment (oerc0122) [reply horace_core/spaghetti_plot.m:202]: The suggested change has a blank line on line 190, so I am not sure what you have copied

**Review by cmarooney-stfc — APPROVED** [review]

File: herbert_core/classes/data_loaders/@loader_ascii/load_data.m
Comment (oerc0122) [inline herbert_core/classes/data_loaders/@loader_ascii/load_data.m:1]: Likewise

File: herbert_core/classes/data_loaders/@nxspepar_loader/private/load_nxspe_par_.m [hunk 2]
```diff
@@ -23,7 +23,7 @@
     end
     if strncmpi(obj.nxspe_version_,'1.1',3)
         warning('LOAD_NXSPE:old_version',...
-            ' you are loading detector data from nxspe data file version 1.1. This nxspe file contains incorrect detectors sizes for rings, so you should use par file for rings');
+            ' you are loading detector data from nxspe data file version 1.1. This nxspe file contains incorrect detectors data for rings, so you should use par file for rings');
```
Comment (oerc0122) [inline herbert_core/classes/data_loaders/@nxspepar_loader/private/load_nxspe_par_.m:26]:
```suggestion
            ' you are loading detector data from nxspe data file version 1.1. This nxspe file contains incorrect detector data for rings, so you should use a par file for rings');
```

File: herbert_core/classes/data_loaders/@nxspepar_loader/private/load_nxspe_par_.m
Comment (oerc0122) [inline herbert_core/classes/data_loaders/@nxspepar_loader/private/load_nxspe_par_.m:1]: This shouldn't be in this PR

File: herbert_core/utilities/general/parse_arguments.m [hunk 3]
```diff
@@ -836,7 +836,12 @@
             ipos=find(strcmpi(args{i},namchk)); % strings in namchk are unique
             if ~isempty(ipos), iskey=true; end
         else
-            ipos=stringmatchi(args{i},namchk);
+            if strncmp(args{i},'-',1)
```
Comment (oerc0122) [inline herbert_core/utilities/general/parse_arguments.m:839]: This seems a somewhat general change to be in this PR

File: horace_core/spaghetti_plot.m [hunk 4]
```diff
@@ -20,6 +20,9 @@
 %
 %   >> [wdisp,cuts] = spaghetti_plot(...)                    % generates a set of 1D cuts
 %   >> [wdisp, cuts] = spaghetti_plot(...,'withpix')         % return cuts as sqw rather than d1ds
+%   >> [...,fig,axes,plots] = spaghetti_plot(...,'withpix')  % in addition
```
Comment (oerc0122) [inline horace_core/spaghetti_plot.m:23]: Is this extra info only available when `'withpix'` is true?

File: horace_core/spaghetti_plot.m [hunk 5]
```diff
@@ -250,11 +261,9 @@
 
     end
 end
-
-
 if ~isempty(plane_normal)
-    fprintf('spaghetti_plot: rlp found to lie in the plane perpendicular to (%g %g %g)\n', ...
-            sqw_proj.transform_img_to_hkl(plane_normal'));
+    fprintf(2,'*** spaghetti_plot: rlp found to lie in the plane perpendicular to (%g %g %g)\n', ...
```
Comment (oerc0122) [inline horace_core/spaghetti_plot.m:265]: Is there a reason this should be printed to (presumably) stderr?

File: horace_core/sqw/@SQWDnDBase/private/cut_parse_inputs_.m
Comment (oerc0122) [inline horace_core/sqw/@SQWDnDBase/private/cut_parse_inputs_.m:1]: Again, these remaining changes should be a separate PR changing `parse_arguments`.

**Review by oerc0122 — CHANGES_REQUESTED** [review]

File: horace_core/spaghetti_plot.m (hunk 5 above)
Comment (abuts) [reply horace_core/spaghetti_plot.m:265]: yes -- generally this is error or warning as spaghetti plot's point do not lie in one plane. If they are, something is usually wrong with your data from scientific point of view Programmatically ok, but if you've bothered to organize  the check for it not to lie in one plane, its probably worth highlighint this in the warning.

**Comment by abuts** [comment]
fixes Re #1649
